ethers-solc/src/resolver/mod.rs

```diff
@@ -903,32 +905,27 @@ impl Node {
     ///
     /// This returns an error if the file's version is invalid semver, or is not available such as
     /// 0.8.20, if the highest available version is `0.8.19`
-    #[allow(dead_code)]
+    #[cfg(all(feature = "svm-solc", not(target_arch = "wasm32")))]
     fn check_available_version(
         &self,
         all_versions: &[SolcVersion],
         offline: bool,
-    ) -> Option<std::result::Result<(), SourceVersionError>> {
-        fn ensure_version(
-            v: &str,
-            all_versions: &[SolcVersion],
-            offline: bool,
-        ) -> std::result::Result<(), SourceVersionError> {
-            let req: VersionReq =
-                v.parse().map_err(|err| SourceVersionError::InvalidVersion(v.to_string(), err))?;
-
-            if !all_versions.iter().any(|v| req.matches(v.as_ref())) {
-                return if offline {
-                    Err(SourceVersionError::NoMatchingVersionOffline(req))
-                } else {
-                    Err(SourceVersionError::NoMatchingVersion(req))
-                }
-            }
+    ) -> std::result::Result<(), SourceVersionError> {
```

Comment:
hmm, why this change?

Reply:
Complicates for no reason, it's a version validation